core/src/main/java/org/everit/json/schema/internal/TemporalFormatValidator.java

```diff
@@ -0,0 +1,39 @@
+package org.everit.json.schema.internal;
+
+import org.everit.json.schema.FormatValidator;
+import org.everit.json.schema.Validator;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Base class for date and time format validators
+ */
+public class TemporalFormatValidator implements FormatValidator {
+    private DateTimeFormatter formatter;
+    private List<String> formats_accepted;
+
+    final static DateTimeFormatter SECONDS_FRACTION_FORMATTER = new DateTimeFormatterBuilder()
+            .appendFraction(ChronoField.NANO_OF_SECOND, 1, 9, true)
+            .toFormatter();
+    static final String ZONE_OFFSET_PATTERN = "XXX";
+
+    TemporalFormatValidator(DateTimeFormatter formatter, List<String> formats_accepted) {
+        this.formatter = formatter;
+        this.formats_accepted = formats_accepted;
```

Comment:
Minor nitpicking here: would you please add a null check for `formatter` (so ` this.formatter = requireNonNull("formatter", "formatter cannot be null";`) and defensively copy the `formatsAccepted` list? (so `this.formatsAccepted = new ArrayList<>(formatsAccepted);`)
Thanks again.